src/CoaddPsf.cc

```diff
@@ -221,6 +221,30 @@ void addToImage(
 }
 
 
+afw::geom::Box2I CoaddPsf::doComputeBBox(
+    afw::geom::Point2D const & ccdXY,
+    afw::image::Color const & color
+) const {
+    afw::table::ExposureCatalog subcat = _catalog.subsetContaining(ccdXY, *_coaddWcs, true);
+    if (subcat.empty()) {
+        throw LSST_EXCEPT(
+            pex::exceptions::InvalidParameterError,
+            (boost::format("Cannot compute BBox at point %s; no input images at that point.")
+             % ccdXY).str());
+    }
+
+    afw::geom::Box2I overallBBox;
+    for (afw::table::ExposureCatalog::const_iterator i = subcat.begin(); i != subcat.end(); ++i) {
```

Comment:
This is very interesting.  I was trying to guess the best choice range declaration from the examples [here](http://en.cppreference.com/w/cpp/language/range-for).  `auto const &` certainly sounds like a better option though, now that I know it exists!